support-workers/src/main/scala/com/gu/support/workers/lambdas/CreateZuoraSubscription.scala

```diff
@@ -44,36 +51,44 @@ class CreateZuoraSubscription(servicesProvider: ServiceProvider = ServiceProvide
     val promotionService = services.promotionService
     val redemptionService = services.redemptionService
     val zuoraService = services.zuoraService
+    val giftCodeGenerator = services.giftCodeGenerator
     val isTestUser = state.user.isTestUser
     val config: ZuoraConfig = services.config.zuoraConfigProvider.get(isTestUser)
-    CreateZuoraSubscription(state, requestInfo, now, today, promotionService, redemptionService, zuoraService, config)
+
+    ifDigitalSubscriptionGiftRedemption(state.product, state.paymentMethod).map(redemptionData =>
+      redeemGift(redemptionData, state.user.id, requestInfo, state, zuoraService, services.catalogService)
+    ).getOrElse(
+      createSubscription(state, requestInfo, now, today, promotionService, redemptionService, zuoraService, giftCodeGenerator, config)
+    )
```

Comment:
I know it's less high level, but pattern match can be clearer when dealing with both sides of option or either - consider:
```suggestion
    maybeDigitalSubscriptionGiftRedemption(state.product, state.paymentMethod) match {
      case Some(redemptionData) =>
        redeemGift(redemptionData, state.user.id, requestInfo, state, zuoraService, services.catalogService)
      case None =>
        createSubscription(state, requestInfo, now, today, promotionService, redemptionService, zuoraService, giftCodeGenerator, config)
    }
```